kyaml/yaml/merge2/scalar_test.go

```diff
@@ -91,15 +91,14 @@ kind: Deployment
 	{description: `remove scalar -- empty in src`,
 		source: `
 kind: Deployment
-field: {}
```

Comment:
So this PR also retains scalar nulls?

Reply:
This null is in input. Because `{}` is not considered as *empty* so only `null` can be used as *empty* in input.